community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -29,43 +48,68 @@
  */
 public class StorePropertyCursor implements PropertyCursor
 {
-    private PropertyStore propertyStore;
-    private StoreStatement storeStatement;
-    private InstanceCache<StorePropertyCursor> instanceCache;
-    private PropertyRecordCursor propertyRecordCursor;
-    private PropertyBlockCursor propertyBlockCursor;
+    private static final long KEY_BITMASK = 0xFFFFFFL;
+    private static final int BITS_BYTE_SIZE = 32;
+    private static final int INTERNAL_BYTE_ARRAY_SIZE = 4096;
+
+    private final PropertyStore propertyStore;
+    private final DynamicStringStore stringStore;
+    private final DynamicArrayStore arrayStore;
+    private final InstanceCache<StorePropertyCursor> instanceCache;
+
+    private long nextPropertyRecordId;
+    private PageCursor cursor;
+
+    private int offsetAtBeginning;
+    private int remainingBlocksToRead;
+    private long header;
+    private PropertyType type;
+    private int keyId;
+
+    private byte[] bytes = new byte[INTERNAL_BYTE_ARRAY_SIZE];
+    private ByteBuffer buffer = ByteBuffer.wrap( bytes ).order( ByteOrder.LITTLE_ENDIAN );
+    private Bits bits = Bits.bits( BITS_BYTE_SIZE );
+
+    private AbstractDynamicStore.DynamicRecordCursor stringRecordCursor;
+    private AbstractDynamicStore.DynamicRecordCursor arrayRecordCursor;
+    private long currentRecordId;
+    private long originalHeader;
 
     public StorePropertyCursor( PropertyStore propertyStore,
```

Comment:
True. Right now it is already tested as part of the other tests (e.g. PropertyIT), but once Cypher and remoting starts using it more explicitly the unit tests need to be fleshed out